File: frontend/src/store/reducers/cluster.ts
import pb from "./../../api/websocket";
import { SET_CLUSTER_INFO } from "../actionTypes";
const initialState: pb.websocket.ClusterInfo = {
  status: "",
  freeMemory: "",
  freeCpu: "",
  totalMemory: "",
  freeRequestCpu: "",
  freeRequestMemory: "",
  totalCpu: "",
  usageMemoryRate: "",
  usageCpuRate: "",
  requestMemoryRate: "",
  requestCpuRate: "",
};

export const selectClusterInfo = (state: {
  cluster: pb.websocket.ClusterInfo;
}) => state.cluster;

export default function cluster(
  state = initialState,
  action: { type: string; info?: pb.websocket.ClusterInfo },
) {
  switch (action.type) {
    case SET_CLUSTER_INFO:
      return { ...state, ...action.info };
    default:
      return state;
  }
}

File: internal/transformer/cluster.go
package transformer

import (
	"github.com/duc-cnzj/mars/api/v6/proto/websocket"
	"github.com/duc-cnzj/mars/v6/internal/biz"
)

// FromClusterInfo 把 biz.ClusterInfo 转换为 websocket ClusterInfo。
func FromClusterInfo(info *biz.ClusterInfo) *websocket.ClusterInfo {
	if info == nil {
		return nil
	}
	return &websocket.ClusterInfo{
		Status:            info.Status,
		FreeMemory:        info.FreeMemory,
		FreeCpu:           info.FreeCpu,
		FreeRequestMemory: info.FreeRequestMemory,
		FreeRequestCpu:    info.FreeRequestCpu,
		TotalMemory:       info.TotalMemory,
		TotalCpu:          info.TotalCpu,
		UsageMemoryRate:   info.UsageMemoryRate,
		UsageCpuRate:      info.UsageCpuRate,
		RequestMemoryRate: info.RequestMemoryRate,
		RequestCpuRate:    info.RequestCpuRate,
	}
}
